### ISSUE TITLE
Uploading SOC with the same owner and identifier should be rejected

### ISSUE THREAD

**Issue by agazso** [ethersphere/bee#1247]
#### Summary
I uploaded two single owner chunks on the same node to the `/soc` endpoint with the same owner and identifier and it was successful.  As per Book of Swarm (pg. 115 footnote 3): 

> If the chunks are uploaded using the same route, the chunk that comes later will be rejected as already known.

#### Steps to reproduce
```sh
% echo '010000000000000000' | xxd -r -p | curl --data-binary @- --no-buffer 'http://localhost:61633/soc/8d3766440f0d7b949a5e32995d09619a7f86e632/0000000000000000000000000000000000000000000000000000000000000000?sig=6c7ea367cce127578597309f90f5c8996ae01e0e288a25ca5fdd82bcfb999e8f31ac14a64cfc28c4984160ed160f5fa3156ec3c38afc0b413e60cb62daf362171c' 
{"reference":"9d453ebb73b2fedaaf44ceddcf7a0aa37f3e3d6453fea5841c31f0ea6d61dc85"}

% echo '010000000000000001' | xxd -r -p | curl --data-binary @- --no-buffer 'http://localhost:61633/soc/8d3766440f0d7b949a5e32995d09619a7f86e632/0000000000000000000000000000000000000000000000000000000000000000?sig=94b5a602b9621bc2ab4e703d7c4e219d2754a7e06f6dbea4b1de528a8c43a1ed443bea80abc54506a7b06408714c92e1f8d8eecde556bfd414d34cc49fa032501c'  
{"reference":"9d453ebb73b2fedaaf44ceddcf7a0aa37f3e3d6453fea5841c31f0ea6d61dc85"}
```

#### Expected behavior
The second command should not succeed but instead should be rejected.

**Comment by Eknir** [comment]
@agazso, could you please do some basic QA on this task and report your outcome here? Question: does bee behave now as expected? Is the returned error useful?

**Comment by agazso** [comment]
> @agazso, could you please do some basic QA on this task and report your outcome here? Question: does bee behave now as expected? Is the returned error useful?

Yes, this is fixed in 0.5.1 and 409 (Conflict) is returned.
